homeassistant/components/linear_garage_door/light.py

```diff
@@ -0,0 +1,133 @@
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import LinearUpdateCoordinator
+
+SCAN_INTERVAL = timedelta(seconds=60)
+PARALLEL_UPDATES = 0
+SUPPORTED_SUBDEVICES = ["Light"]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Linear Garage Door cover."""
+    coordinator: LinearUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    data = coordinator.data
+
+    async_add_entities(
+        LinearLightEntity(
+            device_id=device_id,
+            device_name=data[device_id]["name"],
+            subdevice=subdev,
+            config_entry=config_entry,
+            coordinator=coordinator,
+        )
+        for device_id in data
+        for subdev in data[device_id]["subdevices"]
+        if subdev in SUPPORTED_SUBDEVICES
+    )
+
+
+class LinearLightEntity(CoordinatorEntity[LinearUpdateCoordinator], LightEntity):
+    """Light for Linear devices."""
+
+    _attr_color_mode = ColorMode.BRIGHTNESS
+    _attr_supported_color_modes = {ColorMode.BRIGHTNESS}
+
+    def __init__(
+        self,
+        device_id: str,
+        device_name: str,
+        subdevice: str,
+        config_entry: ConfigEntry,
+        coordinator: LinearUpdateCoordinator,
+    ) -> None:
+        """Initialize the light."""
+        super().__init__(coordinator)
+
+        self._attr_has_entity_name = True
+        self._attr_name = subdevice
+        self._attr_unique_id = f"{device_id}-{subdevice}"
+        self._config_entry = config_entry
+        self._device_id = device_id
+        self._subdevice = subdevice
+        self.device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._device_id)},
+            name=device_name,
+            manufacturer="Linear",
+            model="Garage Door Opener",
+        )
+
+    @property
+    def is_on(self) -> bool:
+        """Return if the light is on or not."""
+        return bool(
+            self.coordinator.data[self._device_id]["subdevices"][self._subdevice][
+                "On_B"
+            ]
+            == "true"
+        )
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness of the light."""
+        return int(
+            int(
+                self.coordinator.data[self._device_id]["subdevices"][self._subdevice][
+                    "On_P"
+                ]
+            )
+            / 100
+            * 255
+        )
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the light."""
+
+        linear = Linear()
+        await linear.login(
+            email=self._config_entry.data["email"],
+            password=self._config_entry.data["password"],
+            device_id=self._config_entry.data["device_id"],
+            client_session=async_get_clientsession(self.hass),
+        )
```

Comment:
This looks a bit ugly tbf. Isn't the client already present in the coordinator?

Reply:
I know, but Linear in itself is very ugly to begin with. Not even their own mobile app is reliable 😅
The client uses WebSockets, but, it's extremely unreliable, and the only real way to get working communication without somehow not receiving messages anymore, is to pretty much connect only when we need to retrieve data, then disconnect. I haven't really found a good way to fetch data other than this, hence this ugly mess 🥲

Reply:
What would you think about moving this to the coordinator anyway? In what other places do you create this?

Reply:
Should just move `linear = Linear()` to the coordinator and then access it from `cover` and `light`?

Reply:
One question i still have, do we need to login every time?

You could make this a property in the coordinator so every time you fetch coördinator.client you get the logged in cliënt

Reply:
Yes, sorry. Every time you connect, you have to send the log-in request. If you stay connected after sending a request or fetching data, it will completely stop responding to requests, which is why I went with this method.

Reply:
Is there something I need to do here or can this be something fixed in the future?

Reply:
Beta has been cut, so let's take the time to make it nice if you don't mind :)